fluss-client/src/main/java/org/apache/fluss/client/write/Sender.java

```diff
@@ -602,4 +607,31 @@ public void initiateClose() {
         accumulator.close();
         running = false;
     }
+
+    private static final class TableKey {
+        private final long tableId;
+        private final TableInfo tableInfo;
+
+        public TableKey(long tableId, TableInfo tableInfo) {
+            this.tableId = tableId;
+            this.tableInfo = tableInfo;
+        }
+
+        @Override
+        public boolean equals(Object o) {
+            if (this == o) {
+                return true;
+            }
+            if (o == null || getClass() != o.getClass()) {
+                return false;
+            }
+            TableKey tableKey = (TableKey) o;
+            return tableId == tableKey.tableId;
+        }
+
+        @Override
+        public int hashCode() {
+            return Objects.hash(tableId);
```

Comment:
The `TableKey` is super hack, because it is used as a map key, but the TableInfo is not considered for comparison. If we want to carry the TableInfo in the Map, it should belong to the values. However, I think we don't need the TableInfo to check log/kv. I made a change for this.